src/FondOfSpryker/Yves/GoogleTagManagerStoreConnector/Expander/StoreDataLayerExpander.php

```diff
@@ -0,0 +1,106 @@
+<?php
+
+namespace FondOfSpryker\Yves\GoogleTagManagerStoreConnector\Expander;
+
+use FondOfSpryker\Shared\GoogleTagManagerStoreConnector\GoogleTagManagerStoreConnectorConstants;
+use FondOfSpryker\Yves\GoogleTagManagerStoreConnector\Dependency\GoogleTagManagerStoreConnectorToStoreClientInterface;
+use FondOfSpryker\Yves\GoogleTagManagerStoreConnector\GoogleTagManagerStoreConnectorConfig;
+
+class StoreDataLayerExpander implements DataLayerExpanderInterface
+{
+    /**
+     * @var \FondOfSpryker\Yves\GoogleTagManagerStoreConnector\Dependency\GoogleTagManagerStoreConnectorToStoreClientInterface
+     */
+    protected $storeClient;
+
+    /**
+     * @var \FondOfSpryker\Yves\GoogleTagManagerStoreConnector\GoogleTagManagerStoreConnectorConfig
+     */
+    protected $config;
+
+    /**
+     * @param \FondOfSpryker\Yves\GoogleTagManagerStoreConnector\Dependency\GoogleTagManagerStoreConnectorToStoreClientInterface $storeClient
+     * @param \FondOfSpryker\Yves\GoogleTagManagerStoreConnector\GoogleTagManagerStoreConnectorConfig $config
+     */
+    public function __construct(
+        GoogleTagManagerStoreConnectorToStoreClientInterface $storeClient,
+        GoogleTagManagerStoreConnectorConfig $config
+    ) {
+        $this->storeClient = $storeClient;
+        $this->config = $config;
+    }
+
+    /**
+     * @param string $page
+     * @param array $twigVariableBag
+     * @param array $dataLayer
+     *
+     * @return array
```

Comment:
by the returned array you could also show the type of objects container from array if it is possible